app/controllers/serviceLearning/routes.py

```diff
@@ -40,27 +40,36 @@ def slcEditProposal(courseID):
         Route for editing proposals, it will fill the form with the data found in the database
         given a courseID.
     """
-    course = Course.get_by_id(courseID)
-    questionData = (CourseQuestion.select().where(CourseQuestion.course == course))
-    questionanswers = [question.questionContent for question in questionData]
-    courseInstructor = CourseInstructor.select().where(CourseInstructor.course == courseID)
-
-    isAllSectionsServiceLearning = ""
-    isPermanentlyDesignated = ""
-
-    if course.isAllSectionsServiceLearning:
-        isAllSectionsServiceLearning = True
-    if course.isPermanentlyDesignated:
-        isPermanentlyDesignated = True
-    terms = selectSurroundingTerms(g.current_term, 0)
-    return render_template('serviceLearning/slcNewProposal.html',
-                                course = course,
-                                questionanswers = questionanswers,
-                                terms = terms,
-                                courseInstructor = courseInstructor,
-                                isAllSectionsServiceLearning = isAllSectionsServiceLearning,
-                                isPermanentlyDesignated = isPermanentlyDesignated,
-                                redirectTarget=getRedirectTarget())
+    if g.current_user.isStudent:
+        abort(403)
+    if g.current_user.isCeltsAdmin or g.current_user.isFaculty:
+        course = Course.get_by_id(courseID)
+        coursestatus = CourseStatus.get_by_id(course.status)
```

Comment:
Is this necessary? The `course` object we are getting has `course.status` which will provide the same object you are re-grabbing for `coursestatus`.

Reply:
fixed